cvmfs/cvmfs_server

```diff
@@ -5077,6 +5063,176 @@ snapshot() {
   return $retcode
 }
 
+__do_all_snapshots() {
+  local separate_logs=0
+  local logrotate_nowarn=0
+  local skip_noninitial=0
+  local log
+  local fullog
+  local repo
+  local repos
+
+  OPTIND=1
+  while getopts "sni" option; do
+    case $option in
+      s)
+        separate_logs=1
+      ;;
+      n)
+        logrotate_nowarn=1
+      ;;
+      i)
+        skip_noninitial=1
+      ;;
+      ?)
+        shift $(($OPTIND-2))
+        usage "Command snapshot -a: Unrecognized option: $1"
+      ;;
+    esac
+  done
+  shift $(($OPTIND-1))
+
+  if [ ! -d /var/log/cvmfs ]; then
+    if ! mkdir /var/log/cvmfs 2>/dev/null; then
+      die "/var/log/cvmfs does not exist and could not create it"
+    fi
+  fi
+  [ -w /var/log/cvmfs ] || die "cannot write to /var/log/cvmfs"
+
+  if [ $logrotate_nowarn == 0 ] && [ ! -f /etc/logrotate.d/cvmfs ]; then
+    cat << EOF
+/etc/logrotate.d/cvmfs does not exist!
+To prevent this error message, create the file or use -n option.
+Suggested content:
+/var/log/cvmfs/*.log {
+    weekly
+    missingok
+    notifempty
+}
+EOF
+    exit 1
+  fi
+
+  if [ $separate_logs == 0 ]; then
+    # write into a temporary file in case more than one is active at the
+    #  same time
+    fulllog=/var/log/cvmfs/snapshots.log
+    log=/tmp/cvmfs_snapshots.$$.log
+    trap "rm -f $log" 0
+    (echo; echo "Logging in $log at `date`") >>$fulllog
+  fi
+
+  # Sort the active repositories by last snapshot time when on local storage.
+  # For non-local, swissknife only supports checking whether a file exists,
+  #  so only check whether non-initial snapshots are being skipped.
+  repos="$(for replica in /etc/cvmfs/repositories.d/*/replica.conf; do
+
+    # get repository information
+    local repodir="${replica%/*}"
+    repo="${repodir##*/}"
+    unset CVMFS_REPLICA_ACTIVE # remove previous setting, default is yes
+    load_repo_config $repo
+
+    if [ "$CVMFS_REPLICA_ACTIVE" = "no" ]; then
+      continue
+    fi
+
+    local upstream=$CVMFS_UPSTREAM_STORAGE
+    local snapshot_time=0
+    if is_local_upstream $upstream; then
+      local storage_dir=$(get_upstream_config $upstream)
+      local snapshot_file=$storage_dir/.cvmfs_last_snapshot
+      if [ -f $snapshot_file ]; then
+        snapshot_time="$(stat --format='%Y' $snapshot_file)"
+      elif [ $skip_noninitial == 1 ]; then
+        continue
+      fi
+    elif [ $skip_noninitial == 1 ]; then
+      if $user_shell "$(__swissknife_cmd) peek -d .cvmfs_last_snapshot -r ${upstream}" | grep -v -q "available"; then
+        continue
+      fi
+    fi
+
+    echo "${snapshot_time}:${repo}"
+
+  done|sort -n|cut -d: -f2)"
+
+  for repo in $repos; do 
+    if [ $separate_logs == 1 ]; then
+      log=/var/log/cvmfs/$repo.log
+    fi
+
+    (
```

Comment:
:) Yeah - note I wasn't happy with the resolution of CVM-810 either!

Anyhow, don't consider this a blocker - but it would be an improvement if you wanted to tackle it.